app/components/Transfers/ensureNoPendingShieldedBalance.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { ComponentType, useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { routerActions } from 'connected-react-router';
+import { Account } from '~/utils/types';
+import { hasPendingShieldedBalanceTransfer } from '~/database/TransactionDao';
+import SimpleErrorModal from '~/components/SimpleErrorModal';
+
+export interface WithAccount {
+    account: Account;
+}
+
+/**
+ * Component that ensures the account does not have have a pending shielded balance transaction.
+ * Requires the component to have prop account containing an Account.
+ */
+export default function ensureNoPendingShieldedBalance<
+    TProps extends WithAccount
+>(Component: ComponentType<TProps>): ComponentType<TProps> {
+    const EnsureNoPendingShieldedBalanceComponent = (props: TProps) => {
+        const { account } = props;
+        const dispatch = useDispatch();
+        const [showError, setShowError] = useState<boolean>(false);
+
+        useEffect(() => {
+            hasPendingShieldedBalanceTransfer(account.address)
+                .then(setShowError)
+                .catch(() => {});
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, [account.address]);
+
+        return (
+            <>
+                <SimpleErrorModal
+                    show={showError}
+                    header="Account has pending transfer impacting shielded balance"
```

Comment:
Is it clear that a **shield** transaction is a transfer? Also, I think we should add some body text as it looks weird without. Perhaps it could say:
```
Please wait for any pending shielding transactions or encrypted transfers to finalize.
```

Reply:
I've added it, but changed it to:

```
Please wait for any pending shield amounts, unshield amounts or shielded transfers to finalize.
```